src/redux/createNetworkMiddleware.js

```diff
@@ -70,11 +71,19 @@ function didComeBackOnline(action, wasConnected) {
   );
 }
 
+function didQueueResume(action, isQueuePaused) {
+  return (
+    action.type === networkActionTypes.CHANGE_QUEUE_SEMAPHORE &&
+    !isQueuePaused &&
```

Comment:
Wouldn't this entail that the queue was paused before?

So the condition would be:

```js
  return (
    action.type === networkActionTypes.CHANGE_QUEUE_SEMAPHORE &&
    isQueuePaused && // shouldn't it be like this?
    action.payload === 'GREEN'
  );
```

Reply:
I believe we're checking that the queue was paused *before* this action is received, since by the time the middleware receives `CHANGE_QUEUE_SEMAPHORE`, `isQueuePaused` hasn't been updated yet. So I think that `!isQueuePaused` is the correct check. Does that sound right?

Reply:
Right, that's exactly my point.

As you are saying, we're checking that the queue was paused before this action is received, so `isQueuePaused === true` at the moment the action arrives to the middleware. The semaphore action indicates that we should `unblock/unpause` the queue.

So for me, `didQueueResume` means that:
- And action of type `action.type === networkActionTypes.CHANGE_QUEUE_SEMAPHORE` is about to be dispatched.
- The queue is currently paused: `isQueuePaused === true`
- The payload is `SemaphoreColor.GREEN`, meaning we should unpause/unblock the queue

Reply:
Done 👍

Reply:
I still don't see the fix @emanueleDiVizio 🤔

Reply:
True, I've thought about it. Not sure how it plays with Flow tho, since we've defined enums for it... Should I just add a SemaphoreColors object without flow?

Reply:
Can confirm fix is in d0393a6

Reply:
Alright done, do we also want to export `SEMAPHORE_COLOR` const?

Reply:
You changed a different function on https://github.com/rgommezz/react-native-offline/commit/d0393a648443970849b1b8d825eee1f6b1c9cd72 😕

Reply:
> True, I've thought about it. Not sure how it plays with Flow tho, since we've defined enums for it... Should I just add a SemaphoreColors object without flow?

Yes, I think that's better. Once we migrate to Typescript in the future, we'll be able to leverage Enums

Reply:
Uh, sorry about that! Will fix this today!